x/ibc-rate-limit/contracts/rate-limiter/src/contract.rs

```diff
@@ -88,6 +88,9 @@ pub struct ChannelFlowResponse {
     pub max: u128,
 }
 
+// TODO: Lets add some more docs for this, namely that channel_value is the total supply of the denom
+// Q: Is an ICS 20 transfer only 1 denom at a time, or does the caller have to split into several
+// calls if its a multi-denom ICS-20 transfer
```

Comment:
highlighting question

Reply:
We're operating at the [SendPacket](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/relay.go#L151) and [RecvPacket](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/ibc_module.go#L174) level on the ibc transfer app (highlighted in the links). Those functions expect the packet to be of type [FungibleTokenPacketData](https://github.com/cosmos/ibc-go/blob/0a887e47318c21dc4668c1056ceae7a7e795a28c/modules/apps/transfer/types/packet.go#L29), which deals with only one denom at a time. 

I'm not sure where in the call-chain we split these for a multi-denom data, but might be worth figuring out how to test those at the go level (maybe [here](https://github.com/osmosis-labs/osmosis/blob/8128b0ca5bdf0a9c1a4ee021c6ed3323c39e26b1/x/ibc-rate-limit/ibc_middleware_test.go), though I don't see helpers for multi-denom transfers in ibctesting or docs for it in the [spec](https://github.com/cosmos/ibc/blob/main/spec/app/ics-020-fungible-token-transfer/README.md#data-structures))

Reply:
Oh cool, then maybe IBC-go doesn't support multi-denom sends

Reply:
ah your definitely right; https://github.com/cosmos/ibc-go/blob/main/proto/ibc/applications/transfer/v1/tx.proto#L28-L29

multi-denom transfers aren't possible